### DISCUSSION TITLE
Crack surface computation in PF fracture analysis.

### DISCUSSION THREAD

**Discussion by selarem** [idaholab/moose#28664] [Q&A Modules: General] [answered]
Hello,
I would like to know how to compute the total crack surface in a phase field fracture analysis. This should inform on the total crack surface and its evolution during the computation. This quantity is to output in a csv file. 
Regards,
Saber

**Comment by GiudGiud** [comment]
@dschwen @hugary1995

**Reply by selarem** [reply, marked as answer]
Solved using TotalFreeEnergy with f_name = F for the total free energy; and with f_name = fracture_energy for the total fracture energy.
Just note that : 

[./elastic_energy_1]
    type = MaterialRealAux
    property = 'elastic_energy'
    variable = elasticlocal_energy
    execute_on = timestep_end
  [../]

and 


[./elastic_energy_2]
    type = TotalFreeEnergy
    variable = total_elastic_energy
    interfacial_vars = 'c'
    kappa_names = 'kappa_op'
    execute_on = 'TIMESTEP_END'
    f_name = elastic_energy
  [../]

give (slight) different outputs.

41.190242813765,41.195224132079 
41.18302553809,41.188742079556 
41.175391675079,41.181772920692 
41.167576021253,41.174590586641 
41.159834429617,41.167426571582

**Comment by selarem** [comment]
The way I am doing it today is to compute  **total_free_energy** and subscript the **total_elastic_energy**. However, I want to check !

**Reply by selarem** [reply]
I found out that the energy computation is wrong and that mechanical equilibrium is not satisfied. In fact, the the fracture energy the (non local term or gradient term) is not considered I with fracture_energy computation, I only have the local term integration ! I would like to have help in how computing the total fracture energy. Regards
